CMakeLists.txt

```diff
@@ -0,0 +1,254 @@
+cmake_minimum_required(VERSION 3.0.2)
+set(CMP0048 NEW)
+
+project(_tango LANGUAGES CXX
+        VERSION 1.0)
+        
+#need to define the version of the library
+set(MAJOR_VERSION "9")
+set(MINOR_VERSION "3")
+set(PATCH_VERSION "1")
```

Comment:
Agree.

@sdebionne you offered some help for PR #176. Could you give me a hand with this one?

Reply:
@tiagocoutinho Sure do! Today is a holiday in France, but I'll look at it tomorrow. If it cannot wait, look at `cmake/project_version.*` in [esrf-bliss/Lima](https://github.com/esrf-bliss/Lima) repo. You need to `include(cmake/project_version.cmake)` right after the project definition.

Reply:
No hurry. Enjoy you holiday :-)

Reply:
@tiagocoutinho See the PR on your fork.

Couple of questions about the `CMakeLists.txt`:

 - Is there a reason to stick with CMake 3.0.2 (5 years old)?
 - Why not use `find_package()` to find Boost?

```
set(Boost_USE_STATIC_LIBS ON)
find_package(Boost COMPONENTS
    python
    REQUIRED
)
```

Using the library should then be just a matter of calling:

```
target_link_libraries(${PROJECT_NAME} PUBLIC Boost::python)
```

 - Same thing for ZMQ?

Tell me if you are interested in these changes.

Reply:
> Is there a reason to stick with CMake 3.0.2 (5 years old)?
To be honest I don't know. I think only @NexeyaSGara can answer this question
> Why not use find_package() to find Boost?

There are more variables at stake:
* boost needs to be statically compiled, 
* for the correct architecture (32/64) (i guess cmake handles this one)
* use the selected python version

I don't know if this can be done "automatically" in CMake.

> set(Boost_USE_STATIC_LIBS ON)
find_package(Boost COMPONENTS
    python
    REQUIRED
)

how does cmake find boost anyway (sorry Cmake is not my strong :-P)

> Same thing for ZMQ?
Actually I don't think it should be necessary to add omniorb and zeromq at all. The only direct dependencies of pytango are boost-python and tango.

> Tell me if you are interested in these changes.
Currently, my main focus is just to be able to bring the work of @NexeyaSGara to the 9.3.1.
There is always room for improvement. I just didn't want you to loose a lot of time with this specially since the unit tests cannot run on windows yet.

Reply:
I thought `LIBRARY_VERSION` was a special variable being used by CMake.
It seems it is not serving any purpose here since the single source of truth for
the PyTango version is in `tango/release.py`. Therefore I propose to simply skip this definition